tests/unit/test_charm.py

```diff
@@ -666,6 +668,8 @@ def test_exec_error_on_client_changed_event_emitted(
 ) -> None:
     caplog.set_level(logging.ERROR)
     harness.set_can_connect(CONTAINER_NAME, True)
+    # needed otherwise self.tracing.is_ready logs an error
+    setup_tempo_relation(harness)
```

Comment:
I think we can just change the check we do on the logs, something like this should work:
```python
    assert f"Exited with code: {err.exit_code}. Stderr: {err.stderr}" in caplog.record_tuples[0]
```

Reply:
Thanks for the tip, but unfortunately that's not the format of the structs.

I used this:
    error_messages = [record[2] for record in caplog.record_tuples]
    assert f"Exited with code: {err.exit_code}. Stderr: {err.stderr}" in error_messages